src/strings.cpp

```diff
@@ -191,18 +181,11 @@ void CopyOutDParam(std::vector<StringParameterBackup> &backup, size_t num)
  * @param backup The backup to check against.
  * @return True when the parameters have changed, otherwise false.
  */
-bool HaveDParamChanged(const std::vector<StringParameterBackup> &backup)
+bool HaveDParamChanged(const std::span<const StringParameterData> backup)
 {
 	bool changed = false;
 	for (size_t i = 0; !changed && i < backup.size(); i++) {
-		bool global_has_string = _global_string_params.GetParamStr(i) != nullptr;
-		if (global_has_string != backup[i].string.has_value()) return true;
-
-		if (global_has_string) {
-			changed = backup[i].string.value() != _global_string_params.GetParamStr(i);
-		} else {
-			changed = backup[i].data != _global_string_params.GetParam(i);
-		}
+		changed = backup[i] != _global_string_params.GetParam(i);
```

Comment:
The whole `changed` variable seems a bit pointless as we're essentially short cutting anyway. Not sure what I might've been thinking when making the function as it was though...
```suggestion
		if (backup[i] != _global_string_params.GetParam(i)) return true;
```
Remove `bool changed` and check in the loop, and change the `return` after the loop to just return `false`?